multitenancy/src/test/java/org/axonframework/extensions/multitenancy/components/deadletterqueue/MultiTenantDeadLetterQueueTest.java

```diff
@@ -0,0 +1,396 @@
+        subject.registerTenant(secondTenant);
+        subject.getTenantSegment(TenantDescriptor.tenantWithId("tenant-send-to"));
+        subject.getTenantSegment(secondTenant);
+
+        deadLetterQueues.forEach(q-> when(q.amountOfSequences()).thenReturn(10L));
+        assertEquals(20, subject.amountOfSequences());
+
+        deadLetterQueues.forEach(q-> verify(q, times(1)).amountOfSequences());
+    }
+
+    @Test
+    void processSingleTenant() {
+        TenantDescriptor secondTenant = TenantDescriptor.tenantWithId("tenant-second-tenant");
+        subject.registerTenant(secondTenant);
+        subject.getTenantSegment(TenantDescriptor.tenantWithId("tenant-send-to"));
+        subject.getTenantSegment(secondTenant);
+
+        new TenantWrappedTransactionManager(NoTransactionManager.INSTANCE,
+                                            secondTenant).fetchInTransaction(() -> subject.process(m -> true, (d) -> Decisions.evict()));
+
+        verify(deadLetterQueues.get(0), times(0)).process(any(),any());
+        verify(deadLetterQueues.get(1), times(1)).process(any(),any());
+    }
+
+    @Test
+    void processSingleAllTenants() {
+        TenantDescriptor secondTenant = TenantDescriptor.tenantWithId("tenant-second-tenant");
+        subject.registerTenant(secondTenant);
+        subject.getTenantSegment(TenantDescriptor.tenantWithId("tenant-send-to"));
+        subject.getTenantSegment(secondTenant);
+
+        deadLetterQueues.forEach(q-> when(q.process(any(),any())).thenReturn(true));
+        assertEquals(true, subject.process(m -> true, (d) -> Decisions.evict()));
+
+        deadLetterQueues.forEach(q-> verify(q, times(1)).process(any(),any()));
+    }
+
+    @Test
+    void clearSingleTenant() {
+        TenantDescriptor secondTenant = TenantDescriptor.tenantWithId("tenant-second-tenant");
+        subject.registerTenant(secondTenant);
+        subject.getTenantSegment(TenantDescriptor.tenantWithId("tenant-send-to"));
+        subject.getTenantSegment(secondTenant);
+
+        new TenantWrappedTransactionManager(NoTransactionManager.INSTANCE,
+                                            secondTenant).fetchInTransaction(() -> {
+                                                subject.clear();
+                                                return null;
+        });
+
+        verify(deadLetterQueues.get(0), times(0)).clear();
+        verify(deadLetterQueues.get(1), times(1)).clear();
+    }
+
+    @Test
+    void clearAllTenants() {
+        TenantDescriptor secondTenant = TenantDescriptor.tenantWithId("tenant-second-tenant");
+        subject.registerTenant(secondTenant);
+        subject.getTenantSegment(TenantDescriptor.tenantWithId("tenant-send-to"));
+        subject.getTenantSegment(secondTenant);
+
+        deadLetterQueues.forEach(q-> doNothing().when(q).clear());
+        subject.clear();
+
+        deadLetterQueues.forEach(q-> verify(q, times(1)).clear());
+    }
```

Comment:
Nit: You could add tests for the builder method validation. (100% coverage!!! :stuck_out_tongue:)